src/tests.rs

```diff
@@ -0,0 +1,320 @@
+            PeaqStorage::add_item(
+            Origin::signed(origin),
+            item_type.to_vec(),
+            item.to_vec()            
+        ),
+        Error::<Test>::ItemTypeAlreadyExists
+    );
+        
+    });
+}
+
+//Test to add item with item type length exceed maximum limit
+#[test]
+fn add_item_type_length_exceeds_limit_test(){
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);
+        
+        let origin = account_key(ACCT);
+        let item = b"item";
+        let tmp = String::from("a").repeat(65);               
+        let item_type = tmp.as_bytes();    
+        
+        assert_noop!(
+            PeaqStorage::add_item(
+                Origin::signed(origin), 
+                item_type.to_vec(), 
+                item.to_vec()
+        ),
+        Error::<Test>::ItemTypeExceedMax64
+    );      
+        
+    });
+}
+
+//Test to add an item with item length exceed maximum limit
+#[test]
+fn add_item_length_exceeds_limit_test(){
+    new_test_ext().execute_with(||{
+        System::set_block_number(1);
+        
+        let origin = account_key(ACCT);
+        let item_type = b"itemType";
+        let tmp = String::from("a").repeat(129);
+        let item = tmp.as_bytes();
+        
+    assert_noop!(
+        PeaqStorage::add_item(
+            Origin::signed(origin), 
+            item_type.to_vec(), 
+            item.to_vec()
+        ),
+        Error::<Test>::ItemExceedMax128
+    );     
+        
+    });
+}
+
+//Test to update an item
+#[test]
+fn update_item_test_ok() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);     
+
+        let origin = account_key(ACCT);
+        let item_type = b"itemType";
+        let  item = b"item";
+        
+        //Add an item
+        assert_ok!(
+            PeaqStorage::add_item(
+                Origin::signed(origin),
+                item_type.to_vec(),
+                item.to_vec()
+            )
+        );       
+
+        //update item
+        assert_ok!(
+            PeaqStorage::update_item(
+                Origin::signed(origin), 
+                item_type.to_vec(), 
+                b"new_item".to_vec()
+            )
+        );
+        
+    });
+}
+
+//Test to update a non existing item
+#[test]
+fn update_non_existing_item_test() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);     
+
+        let origin = account_key(ACCT);
+                
+        assert_noop!(
+            PeaqStorage::update_item(
+                Origin::signed(origin), 
+                b"new_item_type".to_vec(),
+                b"new_item".to_vec()
+            ),
+            Error::<Test>::ItemNotFound
+
+        );              
+
+    });
+}
+
+#[test]
+fn update_item_with_item_length_exceed_limit_test() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);     
+
+        let origin = account_key(ACCT);
+        let item_type = b"itemType";
```

Comment:
How about the item and item_type? Should we extract this?

Reply:
We need to change Item and Item type in some cases.

Reply:
Moved the item and item type to global scope as constants as well.